packages/did-ipfs-store/src/didStore.ts

```diff
@@ -48,6 +61,30 @@ export class DidStore implements IDidStore {
     }
   }
 
+  /**
+   * Checks if file is pinned on cluster
+   * @param cid CID
+   */
+  async isPinned(cid: string): Promise<boolean> {
+    return values((await this.status(cid)).peer_map).some(
+      (pinInfo) => pinInfo.status === TrackerStatus.Pinned
+    );
```

Comment:
How does this pin check work? If only checking `some`, is it possible that it would only be pinned on one node, but not the others? Is that the desired behaviour?

Reply:
Yes, it is possible that data is pinned only on one node, because it is configured by replication factor, which is set in cluster configuration. Probably we can improve this part by waiting enough time to replicate on `replication_factor_min` nodes https://ipfscluster.io/documentation/guides/pinning/#replication-factors. I'll try this. Thanks for notice
It is also possible to set replication factor on adding, but I am not sure if we should expose this option as it requires some knowledge about this process